src/main/kotlin/me/partlysanestudios/partlysaneskies/events/skyblock/dungeons/DungeonEndEvent.kt

```diff
@@ -0,0 +1,19 @@
+package me.partlysanestudios.partlysaneskies.events.skyblock.dungeons
+
+import me.partlysanestudios.partlysaneskies.events.EventManager
+
+class DungeonEndEvent {
+    companion object {
+        internal fun onMessageRecieved(functionList: List<EventManager.EventFunction>, formattedMessage: String) {
+            if (formattedMessage.contains("§r§c☠ §r§eDefeated §r")) {
+                for (function in functionList) {
+                    try {
+                        function.function.call(function.obj, DungeonStartEvent())
```

Comment:
Is there a reason why this is the start event while we are in the end event class?

Reply:
Because I'm stupid

Reply:
Happens